design/specs/listing-views.md

```diff
@@ -1,27 +1,109 @@
 ---
-title: Listing and Searching Views
+title: Listing Views Spec
 description: 
 published: true
-date: 2021-09-17T08:38:09.720Z
+date: 2021-12-20T09:00:00.201Z
 tags: 
 editor: markdown
-dateCreated: 2021-09-17T08:28:58.527Z
+dateCreated: 2021-12-20T09:00:00.201Z
 ---
 
-# Context
-When using Mathesar, users should rely on views for most analysis tasks and tables for more structural tasks like modeling and data transformation. However, both objects are similar and share some functionality. A clear distinction needs to be made between tables and views. Views should also take a more prominent space, and users in a table context should understand that saving as a view will allow them to choose a subset of columns or combine with other tables. 
+## Context
 
-# Prototype
-[Prototype for Listing and Finding Views](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?page-id=5343%3A76906&node-id=5343%3A76907&viewport=338%2C48%2C0.72&scaling=contain&starting-point-node-id=5343%3A76907)
+Database objects are defined objects in a database that store or reference data. In the context of Mathesar, two of those objects are listed and accessible to users as `Tables` and `Views`. Together, these objects are at the core of Mathesar's user experience. One part of that experience is finding and opening these objects and differentiating both.
+
+## Prototype
+
+### Link
+
+[Prototype](https://mathesar-prototype.netlify.app/)
+
+### Videos
+
+- Scenario 1: [Video](https://www.loom.com/share/179070149f00419e9125bd8fba675aa4)
+- Scenario 2: [Video](https://www.loom.com/share/e32500bc744246e5b926852231577fd7)
+- Scenario 3: [Video](https://www.loom.com/share/d553e7edbb28481b8aed02b055270154)
+- Scenario 4: [Video](https://www.loom.com/share/91a20c4770ef4439860425a1bcd84db6)
+
+## User Experience
+
+### Listing of Views and Searching by Name
```

Comment:
@ghislaineguerin 
Tables and views are currently treated as if they are on the same level i.e equal objects to the user, however that's not the case.

The user should be able to identify that views are a higher layer over tables, and tables are more of a raw data store. This should be made evident in both the sidebar and the schema details page.

I'm not sure how we would be able to get this message across effectively, but I'm certain that representing them both on the same level would only get the wrong message across.

@kgodey Do we need to worry about this for the alpha or should we wait until after alpha? I assume it's going to be very tricky to figure out an UX for this.

Reply:
@pavish I had the same comment as you, I think it's important to solve it for the alpha since it's a fundamental concept for the product.

Reply:
@kgodey I saw your comment on separating them out and I agree with it for the alpha, but I also think that we should also find a way to represent the difference in level to the user.

* If we keep them as separate tabs in the schema details page or on the sidebar, they appear like they are equal level objects which are just different from each other.
* We need to represent that they are not equal and views are a higher layer over tables and I think it would require a lot more thought and experimentation. This should be made clear to the user from the very beginning of using the product.

Reply:
I see what you mean. If @ghislaineguerin has a design concept that doesn't take too much work, we can figure it out for the alpha. Otherwise, I think it's fine to defer until afterwards.